### DISCUSSION TITLE
DOCTEST_CONFIG_DISABLE : Test code is still compiled?

### DISCUSSION THREAD

**Discussion by daudrain** [onqtam/doctest#740] [Q&A] [answered]
Hi,

According to the documentation, enabling `DOCTEST_CONFIG_DISABLE` leads to remove everything from the binary.

When the following code exists, the compiler complains about `undeclared identifier foo`, so it looks like test code is still being compiled when `DOCTEST_CONFIG_DISABLE` is enabled:
```
#ifndef DOCTEST_CONFIG_DISABLE
void foo() {
}
#endif
TEST_CASE("FooTest") {
    foo();
}
```

Is it the expected behavior? meaning we still want the code being compiled so build fails if test code is not ok, and we rely on linker to remove all unused functions?

Thank you

**Comment by onqtam** [comment, marked as answer]
Yeah, the code is still being compiled, but it is being turned into a template that is never instantiated. But should still be able to parse. If you want to remove it completely you should #ifdef it altogether just like the function foo
